gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/JwtSecurityInitializer.java

```diff
@@ -176,4 +184,51 @@ public PublicKey getJwtPublicKey() {
         return Optional.of(rsaKey.toPublicJWK());
     }
 
+    /**
+     * Only for unit testing
+     */
+    ZosmfListener getZosmfListener() {
+        return zosmfListener;
+    }
+
+    class ZosmfListener {
+        private boolean isZosmfReady = false;
+        private final ApimlDiscoveryClient discoveryClient;
+
+        private ZosmfListener(ApimlDiscoveryClient discoveryClient) {
+            this.discoveryClient = discoveryClient;
+        }
+
+        // instance variable so can create an accessor for unit testing purposes
+        private final EurekaEventListener zosmfRegisteredListener = new EurekaEventListener() {
+            @Override
+            public void onEvent(EurekaEvent event) {
+                if (event instanceof CacheRefreshedEvent && providers.isZosmfAvailableAndOnline()) {
+                    discoveryClient.unregisterEventListener(this); // only need to see zosmf up once to validate jwt secret
+                    isZosmfReady = true;
+
+                    try {
+                        validateInitializationAgainstZosmf();
+                    } catch (HttpsConfigError e) {
+                        System.exit(1);
```

Comment:
I would strongly suggest adding a log message before killing Gateway

Reply:
`apimlLog.log("org.zowe.apiml.gateway.jwtKeyMissing", keyAlias, keyStore);` will be logged before GW is killed here.

The `HttpsConfigError` on its own will normally kill the GW because it's thrown during bean instantiation, but in this case this isn't a part of bean instantiation and so the error needs to be caught and thrown. Regardless, before that error is thrown the above message is logged.

Reply:
Can we at least be specific and say that this error will cause GW to shut down, similarly to this message?
```    - key: org.zowe.apiml.security.zosmfInstanceNotFound
      number: ZWEAG108
      type: ERROR
      text: "z/OSMF instance '%s' not found or incorrectly configured. Gateway is shutting down."
      reason: "The Gateway could not find the z/OSMF instance from the Discovery Service or it could not communicate with provided z/OSMF instance."
      action: "Ensure that the z/OSMF instance is configured correctly and that it is successfully registered to the Discovery Service and that API Mediation Layer can communicate with provided z/OSMF instance."```